src/styles/_catalog.scss

```diff
@@ -1,22 +1,38 @@
-.catalog {
-  display: flex;
-  flex-wrap: wrap;
+$column-width: 200px;
+$column-height: 300px;
 
-  max-width: 944px;
+.catalog {
+  display: grid;
+  gap: 48px;
+  grid-template-columns: 200px;
+  place-items: center;
+  justify-content: center;
   margin: 0 auto;
+  padding-bottom: 48px;
 
   &__card {
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    width: 200px;
-    height: 300px;
+    display: grid;
+    grid-template-columns: #{$column-width};
+    grid-template-rows: #{$column-height};
+    place-items: center;
+    width: #{$column-width};
+    min-height: #{$column-height};
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media (min-width: 488px) {
+    grid-template-columns: repeat(2, 1fr);
+    grid-template-rows: repeat(4, 1fr);
+  }
+
+  @media (min-width: 768px) {
+      grid-template-columns: repeat(3, 1fr);
+      grid-template-rows: repeat(3, 1fr);
+    }
+
+  @media (min-width: 1024px) {
+    grid-template-columns: repeat(4, 1fr);
+    grid-template-rows: repeat(2, 1fr);
```

Comment:
You dont need to hardcode `grid-template-rows`, can just remove it. Also grid columns not suppose to stretch to 1fr, probably better to set card width for them.